packages/mui-material-next/src/InputAdornment/InputAdornment.tsx

```diff
@@ -41,52 +34,59 @@ const useUtilityClasses = (ownerState) => {
 const InputAdornmentRoot = styled('div', {
   name: 'MuiInputAdornment',
   slot: 'Root',
-  overridesResolver,
-})(({ theme, ownerState }) => ({
-  display: 'flex',
-  height: '0.01em', // Fix IE11 flexbox alignment. To remove at some point.
-  maxHeight: '2em',
-  alignItems: 'center',
-  whiteSpace: 'nowrap',
-  color: (theme.vars || theme).palette.action.active,
-  ...(ownerState.variant === 'filled' && {
-    // Styles applied to the root element if `variant="filled"`.
-    [`&.${inputAdornmentClasses.positionStart}&:not(.${inputAdornmentClasses.hiddenLabel})`]: {
-      marginTop: 16,
-    },
-  }),
-  ...(ownerState.position === 'start' && {
-    // Styles applied to the root element if `position="start"`.
-    marginRight: 8,
-  }),
-  ...(ownerState.position === 'end' && {
-    // Styles applied to the root element if `position="end"`.
-    marginLeft: 8,
-  }),
-  ...(ownerState.disablePointerEvents === true && {
-    // Styles applied to the root element if `disablePointerEvents={true}`.
-    pointerEvents: 'none',
-  }),
-}));
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[`position${capitalize(ownerState.position)}`],
+      ownerState.disablePointerEvents === true && styles.disablePointerEvents,
+      styles[ownerState.variant],
+    ];
+  },
+})<{ ownerState: InputAdornmentOwnerState }>(({ theme, ownerState }) => {
+  const { vars: tokens } = theme;
+  return {
+    '--md-comp-input-adornment-color': tokens.sys.color.onSurfaceVariant,
+    display: 'flex',
+    maxHeight: '2em',
+    alignItems: 'center',
+    whiteSpace: 'nowrap',
+    color: 'var(--md-comp-input-adornment-color)',
+    ...(ownerState.position === 'start' && {
+      // Styles applied to the root element if `position="start"`.
+      marginRight: 16,
```

Comment:
Checking the design guidelines from the design resources of MD3, this distance is actually `12px` instead of `16px`

![image](https://github.com/mui/material-ui/assets/37222944/37926b62-92f4-4521-a112-c33eb8cf6c3c)